### DISCUSSION TITLE
Include a playbar button to change the playback order

### DISCUSSION THREAD

**Discussion by takaji10** [TT-ReBORN/Georgia-ReBORN#17] [Q&A] [answered]
Consider adding a button in the playbar that controls playback order. Default state would be "Default," clicking a second time changes to "Random," clicking again changes to "Repeat Track."

While I do have a number of playlists, one of my favourite ways of exploring my library is using my "All Music" playlist (which is force-sorted and displays all music in my library) and then setting playback on random. When enabling "cursor follows playback," the playlist is always focused on the currently-playing track.

The current "Randomize Playlist" icon in the playbar produces an unexpected action (based on how the majority of playbars operate in foobar2000 and other players) of actually shuffling the order of the tracks in the playlist randomly, which takes some time for foobar2000 to then render in the playlist. This option is made even less helpful when the user has an "All Music" playlist that is force-sorted as it simply doesn't work.

This is how I've tended to use foobar2000 over the years but I'm happy to hear if there's a better way to achieve what I'm looking for?

**Comment by TT-ReBORN** [comment, marked as answer]
Great idea 👍, gonna implement it. 

Actually I never really liked and used the shuffle playlist button. 
Let's make it useful again with redesigning it to the playback order button.

For people who liked this function you can also use it by right click Sort -> Randomize.

-TT

**Reply by takaji10** [reply]
Awesome! Thanks very much - looking forward. :)

**Reply by TT-ReBORN** [reply]
The switchable playback order modes button has been implemented.

-TT

**Reply by takaji10** [reply]
The playback order mode is not being remembered after restarting foobar2000. Is that intentional?

**Reply by TT-ReBORN** [reply]
Yes, I don't want users confused when playback order is not on default (sequential) and they're wondering what's going on when the next song starts to play. It's not much of a burden to click on the playback order button if particular users want to choose another playback order in one foobar session...

Personally I never cared about playback orders, always used sequential...
